server/project.go

```diff
@@ -94,11 +94,11 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 		}
 
 		var activeProjectList []*api.Project
-		for _, project := range projectList {
-			projectList = append(projectList, project)
-			// We will filter those project with the current principle as an inactive member (the role provider differs from that of the project)
-			if projectFind.PrincipalID != nil {
-				principalID := *projectFind.PrincipalID
+		// if principalID is passed, we will enable the filter logic
+		if projectFind.PrincipalID != nil {
+			principalID := *projectFind.PrincipalID
```

Comment:
How about renaming to `userID`, because it's parsed from `c.QueryParam("user")`.

Reply:
Since it's a bug fix, let's minimize the change.